File: ruby/resistor-color-duo/resistor_color_duo.rb
=begin
Write your code for the 'Resistor Color Duo' exercise in this file. Make the tests in
`resistor_color_duo_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/resistor-color-duo` directory.
=end

class ResistorColorDuo
  COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]

  def self.value(arr_duo_resistor)
    first = COLORS.find_index { |color| color == arr_duo_resistor[0] }
    second = COLORS.find_index { |color| color == arr_duo_resistor[1] }
    "#{first}#{second}".to_i
  end
end

File: javascript/resistor-color-duo/resistor-color-duo.js
export const decodedValue = ([resistor_color_1, resistor_color_2]) => {
  const first = COLORS.findIndex((color) => color === resistor_color_1);
  const second = COLORS.findIndex((color) => color === resistor_color_2);
  return parseInt(`${first}${second}`);
};

const COLORS = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"];